scripts/commands/gotonpc.lua

```diff
@@ -0,0 +1,84 @@
+-----------------------------------
+-- func: gotonpc
+-- desc: Go to given npc by name
+-----------------------------------
+require("scripts/globals/status")
+
+cmdprops =
+{
+    permission = 1,
+    parameters = "si"
+}
+
+function error(player, msg)
+    player:PrintToPlayer(msg)
+    player:PrintToPlayer("!gotonpc <npcName> (index)")
+end
+
+local removeInvalidNpcs = function(npcs)
+    local validNpcs = {}
+    for i, npc in pairs(npcs) do
+        local pos = npc:getPos()
+        local pos0 = (pos.x == 0 and pos.y == 0 and pos.z == 0)
+
+        -- only add npc if it is at valid coordinates
+        if not pos0 then
+            table.insert(validNpcs, npc)
+        end
+    end
+
+    return validNpcs
+end
+
+local goToNpc = function(player, npc)
+    -- determine whether we need zoneId parameter
+    local gotoZone = nil
+    if npc:getZoneID() ~= player:getZoneID() then
+        gotoZone = npc:getZoneID()
+    end
+
+    -- display message
+    player:PrintToPlayer(string.format("Going to %s %s (%i).", npc:getName(), npc:getZoneName(), npc:getID()))
+
+    -- half a second later, go.  this delay gives time for previous message to appear
+    player:timer(500, function(playerArg)
+        playerArg:setPos(npc:getXPos(), npc:getYPos(), npc:getZPos(), npc:getRotPos(), gotoZone)
+    end)
+end
+
+function onTrigger(player, npcName, index)
+    -- validate npc
+    if npcName == nil or npcName == "" then
+        error(player, "You must enter an NPC name")
+        return
+    end
+
+    -- get and filter npcs by name
+    local npcs = GetNPCsByName(npcName)
```

Comment:
If you used `zone:queryEntitiesByName(...)` you wouldn't have to create a new binding, and you could probably re-implement your `matches' algorithm in Lua (it's just binary search on collections of less-than-32, so its not a big deal either way

Reply:
I can try how it would look like. Is there a binding or method of obtaining all zones in lua?

Reply:
Had a discussion with @jmcmorris on discord. Since `zone:queryEntitiesByName` only works in a single zone, I think the best route here if we want to reuse that method would be to create a new binding in luautils `QueryEntitiesByName` that calls through `zone:queryEntitiesByName`. I think it would be valuable to add pattern matching to the original `zone:queryEntitiesByName` method since it should be backwards compatible and would make our new code work without many lua changes

Reply:
Reused `zone:queryEntitiesByName` and limited search to only current zone